### PR TITLE
Video: Add captions + controls

### PR DESCRIPTION
**Pull request by jesselumarie**
Currently, the `captions` attribute on videos doesn't actually show captions on a video, we need to update the video text track state to show the captions.

## Test Plan
  - [x] verify that captions auto-play when provided
  - [x] verify that captions stop when toggled off
  - [x] verify captions restart when toggled on

captions

### CODE REVIEW

File: packages/gestalt/src/Video.js [hunk 1]
```diff
@@ -20,7 +20,6 @@ type Props = {|
   accessibilityUnmuteLabel: string,
   aspectRatio: number,
   captions: string,
-  children?: React.Node,
```
Comment (christianvuerings) [inline packages/gestalt/src/Video.js:23]: @jesselumarie what's the reason for removing this line? Could we add it back in?

image

Comment (jesselumarie) [reply packages/gestalt/src/Video.js:23]: definite rebase mistake -- good catch

File: packages/gestalt/src/Video.js [hunk 2]
```diff
@@ -331,6 +337,25 @@ export default class Video extends React.PureComponent<Props, State> {
     }
   };
 
+  // Toggle captions on/off
+  toggleCaptions: () => void = () => {
+    if (this.video && this.video.textTracks && this.video.textTracks[0]) {
```
Comment (jackhsu978) [inline packages/gestalt/src/Video.js:342]: nit: can we simplify the code here a little?

```js
toggleCaptions: () => {
  const [videoTrack] = this.video?.textTracks || [];
  if (videoTrack) {
    const isShowing = videoTrack.mode === 'showing';
    videoTrack.mode = isShowing ? 'disabled' : 'showing';
    this.setState({ captionsButton: isShowing ? 'disabled' : 'enabled' });
  }
};
```

File: packages/gestalt/src/Video.js [hunk 3]
```diff
@@ -236,6 +238,10 @@ export default class Video extends React.PureComponent<Props, State> {
     if (playing) {
       this.play();
     }
+
+    if (this.video && this.video.textTracks && this.video.textTracks[0]) {
```
Comment (jackhsu978) [inline packages/gestalt/src/Video.js:242]: nit: do we need to check the `captions` prop here?

File: packages/gestalt/src/Video.js (hunk 2 above)
Comment (jackhsu978) [reply packages/gestalt/src/Video.js]: lgtm

File: packages/gestalt/src/Video.js [hunk 4]
```diff
@@ -499,6 +520,7 @@ export default class Video extends React.PureComponent<Props, State> {
         <ColorSchemeProvider id="Video" colorScheme="light">
           <video
             autoPlay={playing}
+            crossOrigin="anonymous"
```
Comment (jackhsu978) [inline packages/gestalt/src/Video.js:523]: why is this change necessary?

File: packages/gestalt/src/VideoControls.js [hunk 5]
```diff
@@ -154,6 +174,20 @@ class VideoControls extends React.Component<Props> {
             />
           </TapArea>
         </Box>
+        {captionsButton && (
+          <Box padding={2}>
+            <TapArea onTap={this.handleCaptionsChange} fullWidth={false}>
+              <Icon
+                accessibilityLabel="closed-captions"
```
Comment (christianvuerings) [inline packages/gestalt/src/VideoControls.js:181]: @jesselumarie this should be 2 props that are passed in and it should depend on whether we show / hide the captions.

Something along these lines:
```
<Icon accessibilityLabel={captionsButton === 'enabled' ? accessibilityHideCaptionsLabel : accessibilityShowCaptionsLabel} />
```

File: packages/gestalt/src/Video.js (hunk 4 above)
Comment (jesselumarie) [reply packages/gestalt/src/Video.js:527]: the track needs to match the video origin -- if it doesn't, you need to pass `anonymous`, otherwise the browser will block the captions track from being loaded.
